src/python/pants/python/python_setup.py

```diff
@@ -50,6 +50,24 @@ def register_options(cls, register):
                 "on the format of constraint files and how constraints are applied in Pex and Pip."
             ),
         )
+        register(
+            "--use-all-requirements",
```

Comment:
Maybe `always_use_all_requirements`? or `use_single_resolve`? Without the `always` part, I might think "Why wouldn't you use them all? It sounds like this would error without this?"

Reply:
I'll chew on these. `--use-shared-global-resolve`?

Reply:
Yeah, I like that more. Or `--use-single-global-resolve`?

Reply:
I'm going with `--resolve-all-constraints`, to emphasize that it's tied to the constraints file.